### PR TITLE
Stomatogastric ganglion crab lobster

### PR DESCRIPTION
**Pull request by stevewds**
Hi - 

Here we add parts of the stomatogastric nervous system of crabs and lobsters.

1) We added a new "stomatogastric nervous system" term to cover both crabs and lobsters and insects. We updated the old term "insect stomatogastric nervous system" to be "part of" the stomatogastric nervous system. 

2) We added some 14 parts of the stomatogastric nervous system of crabs and lobsters.

We discussed these changes at the November Uberon meeting.

Thanks
Dre and Steve

### CODE REVIEW

**Comment by stevewds** [comment]
Requesting @rays22 as a reviewer as discussed in the November meeting.

Best wishes
Steve

**Comment by stevevanhooser** [comment]
Thanks @rays22, revised as requested

Best
Steve

File: src/ontology/uberon-edit.obo [hunk 1]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
 relationship: dc-contributor https://orcid.org/0000-0003-2473-2313
 property_value: dcterms-date "2024-06-12T13:51:05Z" xsd:string
 
+[Term]
+id: UBERON:8910000
+name: stomatogastric nervous system
+def: "The part of the nervous system that controls the stomach, such as in crabs, lobsters, and flies" [FlyBase:FBrf0089570, PMID:12966498, PMID:�#27450880, DOI:10.1016/B0-12-370878-8/00177-4]
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224757]: There is a strange character in `PMID:�#27450880`.

File: src/ontology/uberon-edit.obo [hunk 2]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+is_a: UBERON:0011216 ! organ system subdivision
+relationship: in_taxon NCBITaxon:6656 ! Arthropoda
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+
+[Term]
+id: UBERON:8910001
+name: stomatogastric ganglion
+def: "The stomatogastric ganglion (STG) consists of #�30 neurons that form two central pattern generator circuits in crustaceans. STG neurons have large soma (~50-100 um) and have complex branches. The crustacean stomach is a complex mechanical structure that grinds and filters foods. The STG contains the motor neurons that innervate the striated muscles that move the gastric mill and pyloric regions of the stomach." [PMID: 17009928]
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224765]: non-standard character: #�30

File: src/ontology/uberon-edit.obo [hunk 3]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+
+[Term]
+id: UBERON:8910015
+name: pyloric dilator nerve (sensu Cancer borealis)
+def: "Innervates muscles that dilate valves of the pylorus #�� motor neurons involved in filtering in crustaceans. Anterior burster neurons are intrinsic oscillators that are electrically coupled with pyloric dilator neurons, causing them to fire bursts of action potentials. Together these neurons inhibit lateral pyloric and pyloric neurons, forcing them to fire in alternation with the pyloric dilator neurons." [PMID: 2876096, PMID: 17009928]
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224840]: Non-standard character in "pylorus #�� motor".

File: src/ontology/uberon-edit.obo [hunk 4]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+
+[Term]
+id: UBERON:8910020
+name: commissural ganglion (sensu Cancer borealis)
+synonym: "CG (sensu Cancer borealis)"
+def: "Contains motor fibers that innervate the esophagus. Stimulating the pyloric suppressor neuron alters both pyloric and gastric rhythms but also elicits activity in a third oscillatory circuit located in the commissural ganglia called the esophageal rhythm." [DOI:10.1016/b978-0-12-809324-5.21166-8]�#
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224893]: Non-standard character at the end of line.

File: src/ontology/uberon-edit.obo [hunk 5]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+
+[Term]
+id: UBERON:8910021
+name: junction of superior esophageal nerve and stomatogastric nerve (sensu Cancer borealis)
+def: "The junction between the stomatogastric nerve (STN) and the superior esophageal nerve (SON) shows synaptic structures. Octopamine may be released at this site by projection neurons #�� its application at this junction induces action potentials in the axons of the modulatory commissural neuron 5 (MCN5) that project from the commissural ganglia to the stomatogastric ganglion (STG)." [PMID:�#15306640]�#
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224903]: Non-standard characters.

**Review by rays22 — CHANGES_REQUESTED** [review]
There are several non-standard characters in the `uberon-edit.obo` file. Please, remove them.
I would like to check the inferred relationships between the existing and the newly added terms, but Protege fails to open `uberon-edit.obo` from this github branch.
The error message in Protege is `org.xml.sax.SAXParseException`. Could the failure of Protege to parse `uberon-edit.obo` be related to the non-standard characters in the file?

**Comment by stevevanhooser** [comment]
@rays22 , sorry for that, I think I have removed them all. (I don't have a "special character detector" program but I saw the ones above.)

File: src/ontology/uberon-edit.obo [hunk 6]
```diff
@@ -219794,6 +219794,7 @@ synonym: "stomodeal nervous system" RELATED [FlyBase:FBrf0111704]
 is_a: UBERON:0011216 ! organ system subdivision
 relationship: in_taxon NCBITaxon:6656 ! Arthropoda
 relationship: part_of UBERON:0001017 ! central nervous system
+relationship: is_a UBERON:8910000 ! stomatogastic nervous system
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:219797]:
```suggestion
is_a: UBERON:8910000 ! stomatogastric nervous system
```
Fix the syntax for SubClass assertion (and typo in label).

File: src/ontology/uberon-edit.obo [hunk 7]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+def: "The part of the nervous system that controls the stomach, such as in crabs, lobsters, and flies" [FlyBase:FBrf0089570, PMID:12966498, PMID:27450880, DOI:10.1016/B0-12-370878-8/00177-4]
+is_a: UBERON:0011216 ! organ system subdivision
+relationship: in_taxon NCBITaxon:6656 ! Arthropoda
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+
+[Term]
+id: UBERON:8910001
+name: stomatogastric ganglion
+def: "The stomatogastric ganglion (STG) consists of about 30 neurons that form two central pattern generator circuits in crustaceans. STG neurons have large soma (~50-100 um) and have complex branches. The crustacean stomach is a complex mechanical structure that grinds and filters foods. The STG contains the motor neurons that innervate the striated muscles that move the gastric mill and pyloric regions of the stomach." [PMID: 17009928] 
+synonym: "STG"
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224766]:
```suggestion
synonym: "STG" EXACT []
```
Fix incorrect syntax.

File: src/ontology/uberon-edit.obo [hunk 8]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+def: "Consists of lateral gastric (LG) neurons. The alternation between the lateral gastric neurons/medial gastric neurons and the lateral posterior gastric neurons control the movements of the lateral teeth in crustaceans." [PMID: 17009928]
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+relationship: part_of UBERON:8910001 ! stomatogastric ganglion
+
+[Term]
+id: UBERON:8910013
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224818]: Use unique IDs. UBERON:8910013 has been used twice.

**Comment by stevevanhooser** [comment]
Thanks @rays22, fixed and updated

File: src/ontology/uberon-edit.obo [hunk 9]
```diff
@@ -219794,6 +219794,7 @@ synonym: "stomodeal nervous system" RELATED [FlyBase:FBrf0111704]
 is_a: UBERON:0011216 ! organ system subdivision
 relationship: in_taxon NCBITaxon:6656 ! Arthropoda
 relationship: part_of UBERON:0001017 ! central nervous system
+relationship: is_a UBERON:8910000 ! stomatogastric nervous system
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:219797]:
```suggestion
is_a: UBERON:8910000 ! stomatogastric nervous system
```
The syntax for this SubClassOf  assertion still does not look correct. You can see what I mean If you open `uberon-edit.obo` in Protege and check if the "insect stomatogastric nervous system" is a sub-class of "stomatogastric nervous system".

**Review by rays22 — CHANGES_REQUESTED** [review]

**Comment by stevevanhooser** [comment]
Thanks @rays22 . 

I'm a little confused. I've opened this in protoge, and I see it as a subclass, but I may not be viewing it the same way that you are. Sorry for my newbie issues!

Here if I'm looking at insect stomatogastric nervous system in Protoge it says it "is a some stomatogastric nervous system"

Screenshot 2024-12-17 at 1 39 23 PM

**Comment by rays22** [comment]
> Thanks @rays22 .
> 
> I'm a little confused. I've opened this in protoge, and I see it as a subclass, but I may not be viewing it the same way that you are. Sorry for my newbie issues!
> 
> Here if I'm looking at insect stomatogastric nervous system in Protoge it says it "is a some stomatogastric nervous system"
> 
> Screenshot 2024-12-17 at 1 39 23 PM

It should look like this:
subClassOf-somatogastric_nervous_system

1. The `insect somatogastric nervous system` should be a branch of the `somatogastrcic nervous system` in the Class Hierarchy (left arrow).
2. The `is_a some 'somatogastric nervous system'` is a slightly but significantly different axiom from what we want here (right arrow).

**Comment by stevevanhooser** [comment]
Thanks Ray. Do you know how I should fix it? I'm not sure which language the .obo file is.

**Comment by stevevanhooser** [comment]
All tests pass after merging as well

**Comment by rays22** [comment]
> Thanks Ray. Do you know how I should fix it? I'm not sure which language the .obo file is.

You can replace line 219797 with my suggested fix. (You might be able to [apply the suggested change directly here in the PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) .

https://github.com/obophenotype/uberon/pull/3445/files#r1889024567

**Comment by stevevanhooser** [comment]
Ah, got it. Sorry, I didn't realize the word "relationship" messed it all up. It's now fixed and looks the same for me as it does for you in Protoge

Thanks
Steve

File: src/ontology/uberon-edit.obo [hunk 10]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+[Term]
+id: UBERON:8910011
+name: dorsal gastric nerve (sensu Cancer borealis)
+def: "Nerve carrying dorsal gastric (DG) neurons which control the medial tooth movements in crustaceans." [PMID: 17009928]
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224793]:
```suggestion
relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```
Keep only one of the two axioms.
* Explanation:
If the `stomatogastric ganglion` is part of the `stomatogastric nervous system`, and another anatomical entity is part of the `stomatogastric ganglion`, then it is redundant to assert that the anatomical entity is also part of the `stomatogastric nervous system`. (The `part of` relationship is transitive.) Multiple classifications chains resulting from redundant assertions make the ontology difficult to edit and maintain in the long run.

File: src/ontology/uberon-edit.obo [hunk 11]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+[Term]
+id: UBERON:8910012
+name: gastropyloric nerve (sensu Cancer borealis)
+def: "Consists of gastropyloric receptor (GPR) neurons, which are stretch receptors that innervate stomach muscles in crustaceans." [PMID: 17009928, PMID: 26888106]
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224804]:
```suggestion
relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```

File: src/ontology/uberon-edit.obo [hunk 12]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+[Term]
+id: UBERON:8910013
+name: lateral gastric nerve (sensu Cancer borealis)
+def: "Consists of lateral gastric (LG) neurons. The alternation between the lateral gastric neurons/medial gastric neurons and the lateral posterior gastric neurons control the movements of the lateral teeth in crustaceans." [PMID: 17009928]
+is_a: UBERON:0035014 ! functional part of brain
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+relationship: in_taxon NCBITaxon:6712 ! Astacidea
+relationship: in_taxon NCBITaxon:6752 ! Brachyura
+relationship: part_of UBERON:8910000 ! stomatogastric nervous system
+relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```
Comment (rays22) [inline src/ontology/uberon-edit.obo:224815]:
```suggestion
relationship: part_of UBERON:8910001 ! stomatogastric ganglion
```

**Review by rays22 — CHANGES_REQUESTED** [review]
Sorry, I have some more (hopefully final) comments. We are getting there ... :smiley:

File: src/ontology/uberon-edit.obo (hunk 10 above)
Comment (stevevanhooser) [reply src/ontology/uberon-edit.obo:224793]: This is an interesting question we should unpack further. Many of these nerves are partially comprised of processes that originate from the stomatogastric ganglion, but then they extend far outside the ganglion around other parts of the stomatogastric nervous system. If you cut out the stomatogastric ganglion, many of these processes would still be there and would still be part of the stomatogastric nervous system.

Is there a different way we should capture the relationship?

This comment is similar for the other cases...

File: src/ontology/uberon-edit.obo [hunk 13]
```diff
@@ -224750,6 +224751,173 @@ is_a: UBERON:0012449 ! mechanoreceptor
...
+name: stomatogastric nervous system
+def: "The part of the nervous system that controls the stomach, such as in crabs, lobsters, and flies" [FlyBase:FBrf0089570, PMID:12966498, PMID:27450880, DOI:10.1016/B0-12-370878-8/00177-4]
+is_a: UBERON:0011216 ! organ system subdivision
+relationship: in_taxon NCBITaxon:6656 ! Arthropoda
+relationship: dc-contributor https://orcid.org/0000-0002-1909-7004
+
+[Term]
+id: UBERON:8910001
+name: stomatogastric ganglion
+def: "The stomatogastric ganglion (STG) consists of about 30 neurons that form two central pattern generator circuits in crustaceans. STG neurons have large soma (~50-100 um) and have complex branches. The crustacean stomach is a complex mechanical structure that grinds and filters foods. The STG contains the motor neurons that innervate the striated muscles that move the gastric mill and pyloric regions of the stomach." [PMID: 17009928]
```
[the remark this answers is not in the document]
Comment (stevevanhooser) [reply src/ontology/uberon-edit.obo:224765]: Makes sense, I will make this change (I can't seem to do it right here in github but I can push a change)

File: src/ontology/uberon-edit.obo (hunk 10 above)
Comment (rays22) [reply src/ontology/uberon-edit.obo:224793]: > This is an interesting question we should unpack further. Many of these nerves are partially comprised of processes that originate from the stomatogastric ganglion, but then they extend far outside the ganglion around other parts of the stomatogastric nervous system. If you cut out the stomatogastric ganglion, many of these processes would still be there and would still be part of the stomatogastric nervous system.
> 
> Is there a different way we should capture the relationship?
> 
> This comment is similar for the other cases...

I see. There is the [RO:0002131 overlaps](https://www.ebi.ac.uk/ols4/ontologies/ro/properties/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FRO_0002131?lang=en) relationship that might  capture this.

**Comment by stevevanhooser** [comment]
Yes, `overlaps` works. Changed!

**Review by rays22 — APPROVED** [review]
Looks good.

**Pull request merged**